source/mainnet/smart-contracts/guides/setup-contract.rst

```diff
@@ -6,6 +6,38 @@
 Setting up a smart contract project
 ===================================
 
+This guide documents two different options (`from a template` or `from scratch`) to create a new Concordium smart contract project.
+The `from a template` option provides you with some
+smart contract templates. Choose the template that best fits your project scope.
+The `from scratch` option guides you through the process when you want to start a new project without any boilerplate code.
+
+.. note::
+   We recommend that newcomers choose the `from a template` option and select the ``default`` template to start with.
+
+From a template:
+================
+
+Concordium maintains several smart contract templates (currently a ``default`` template and a ``cis2-nft`` template).
+To start a new Concordium smart contract project from a template, run the command:
+
+.. code-block:: console
+
+   $cargo concordium init
+
+This command generates a new project from the templates in the
+`template folder <https://github.com/Concordium/concordium-rust-smart-contracts>`_.
+The path where the project should be created can be provided with the ``--path`` option.
+The ``cargo-generate`` crate is required for running the above command. ``cargo-generate`` can
```

Comment:
I think it would be good to state this upfront, some users will just run the command first and fail, before they read until this point :)